pkg/cli/cvd.go

```diff
@@ -87,29 +91,54 @@ func newCVDCommand(opts *subCommandOpts) *cobra.Command {
 func createCVD(c *cobra.Command, flags *CreateCVDFlags, opts *subCommandOpts) error {
 	service, err := opts.ServiceBuilder(flags.CommonSubcmdFlags, c)
 	if err != nil {
-		return err
+		return fmt.Errorf("Failed to build service instance: %w", err)
+
 	}
-	if flags.LocalImage {
-		return createCVDFromLocalBuild(service, c, flags)
+	action := &createCVDAction{
+		Service: service,
+		Opts:    *flags.CreateCVDOpts,
+	}
+	cvd, err := action.Execute()
+	if err != nil {
+		return fmt.Errorf("Failed to create cvd: %w", err)
+	}
+	output := CVDOutput{
+		BaseURL: buildBaseURL(flags.CVDRemoteFlags),
+		Host:    flags.Host,
+		CVD:     cvd,
+	}
+	c.Printf("%s\n", output.String())
+	return nil
+}
+
+type createCVDAction struct {
+	Service client.Service
+	Opts    CreateCVDOpts
+}
```

Comment:
I like the idea of the extra class to break the operation into smaller pieces, but I think you could go even further by having:
* A `CVDCreator` interface with a `Create` method
* A `CVDCreator` builder (class, function, whatever makes the most sense) that parses the flags and instantiates the appropriate `CVDCreator` implementation
* Different implementations of the `CVDCreator` interface for local builds and android ci builds

Reply:
agree, however as of right now the creation logic is quite simple, so having a new interface and two different creator classes seems overwhelming to me. I'll keep this in mind if at least of the creators logic starts getting bigger.